### DISCUSSION TITLE
How should I structure my Xamarin project for Premium and lite version.

### DISCUSSION THREAD

**Discussion by mfaizey** [xamarin/Xamarin.Forms#15441] [General]
Hey everyone, I am developing a xamarin application, I want to run ads on the lite version and the premium version does not contain ads and has some extra features.
My question is should I be creating two separate xamarin projects one for lite and one for premium. or is there any way I can do that in one project?

Any help would be much appreciated!

**Comment by heathbm** [comment]
You should not need to create 2 separate projects. You could use the feature flag pattern:
- https://docs.microsoft.com/en-us/dotnet/architecture/cloud-native/feature-flags
- https://microsoft.github.io/PartsUnlimited/advanced/FeatureFlagMobile.html (excluding the settings UI component)

Once you have done this, you can set the flag, build and release. Then flip the flag and repeat for the other version.

**Reply by mfaizey** [reply]
Yes, I understand. But I am running ads in lite and subscriptions for the other one.

**Reply by heathbm** [reply]
Say the flag is called `IsAppLite`. For ads, you need to manually load them generally. So, you could do the following:

```
if (IsAppLite)
{
    interstitialAd.LoadAd();
}
```

However, this is an implementation detail, and is not strictly related to Xamarin. I would suggest asking for more details in the Xamarin Discord server: https://discord.com/invite/fB6yfFhrX9

**Reply by heathbm** [reply]
@Muhammadfaizann Do you have any further questions regarding this topic?
